packages/typescript-estree/src/create-program/createWatchProgram.ts

```diff
@@ -314,6 +314,10 @@ function createWatchProgram(
       );
     oldOnDirectoryStructureHostCreate(host);
   };
+  // This works only on 3.9
+  (watchCompilerHost as any).extraFileExtensions = extra.extraFileExtensions.map(
```

Comment:
could you please add this property to our custom type?
we can remove this when we bump to 3.9

https://github.com/typescript-eslint/typescript-eslint/blob/9d53c761983dd964109b9f13eb9bfe20caf9defb/packages/typescript-estree/src/create-program/WatchCompilerHostOfConfigFile.ts#L30-L35